Assets/Scripts/Framework/TriggerArea/TriggerAreaBase.cs

```diff
@@ -107,9 +107,9 @@ private protected void ProcessNewCollidingTargets()
                 if (newColliding[i] != null)
                     AddNewComponent(newColliding[i]);
 
-            int collidersCount = _colliders.Count;
+            int collidersLastIndex = _colliders.Count - 1;
 
-            for (int i = 0; i < collidersCount; i++)
+            for (int i = collidersLastIndex; i > 0; i--)
```

Comment:
This should be higher than or equal to.
```suggestion
            for (int i = collidersLastIndex; i >= 0; i--)
```